src/watcher.js

```diff
@@ -266,6 +271,29 @@ class Watcher extends SafeEventEmitter {
         this.on('load.chat', updateChannel);
         this.on('load.vod', updateChannel);
     }
+
+    directoryObserver() {
+        const observe = (watcher, element) => {
+            if (!element) return;
+            if (watcher) watcher.disconnect();
+            watcher.observe(element, {childList: true, subtree: true});
+        };
+
+        directoryWatcher = new window.MutationObserver(mutations =>
+            mutations.forEach(mutation => {
+                for (const el of mutation.addedNodes) {
+                    const $el = $(el);
+
+                    const vodCasts = $el.find('.tw-pill');
+
+                    if (vodCasts.length > 0) {
+                        this.emit('directory.vodcast');
+                    }
+                }
+            })
+        );
+        this.on('load.directory.following', () => observe(directoryWatcher, $('.twilight-main')[0]));
```

Comment:
twilight-main is not scoped enough .. a page change should break the observer such that it isn't constantly watching this dom node.

Reply:
Found and changed a closer element that's removed when you navigate away from the directory area: 'main .scrollable-area .pd-3'